src/components/AppLayout/PlatformSwitcher/PlatformSwitcherButton.tsx

```diff
@@ -0,0 +1,50 @@
+import { ComponentProps, ReactNode } from "react";
+import clsx from "clsx";
+import { LegacyChevronDown2pxIcon } from "@deriv/quill-icons";
+import "./PlatformSwitcher.scss";
+
+type TPlatformSwitcherButton = ComponentProps<"button"> & {
+    icon: ReactNode;
+    isExpanded: boolean;
+    chevronIconSize?: number;
+    chevronIconClassName?: string;
+};
+
+/**
+ * PlatformSwitcherButton component renders a button with an icon and a chevron that indicates expansion state.
+ * @param {TPlatformSwitcherButton} props - The properties passed to the component.
+ * @property {ReactNode} icon - The icon to be displayed inside the button.
+ * @property {boolean} isExpanded - Flag to determine if the switcher is expanded.
+ * @property {number} [chevronIconSize=16] - Optional size for the chevron icon.
+ * @property {string} [chevronIconClassName] - Optional additional class names for the chevron icon.
+ * @returns {JSX.Element} The PlatformSwitcherButton component.
+ */
+export const PlatformSwitcherButton = ({
+    icon,
+    className,
+    chevronIconSize = 16,
+    chevronIconClassName,
```

Comment:
```suggestion
    iconSize = 16,
    iconClassName,
```

just naming icon is better and more generic or is there a specific reason to add `chevron`?

Reply:
since we are using chevron icon I named it like that
but agree with you will rename it 🙏